src/Language/PureScript/Sugar/CaseDeclarations.hs

```diff
@@ -120,10 +120,26 @@ toTuple _ = error "Not a value declaration"
 makeCaseDeclaration :: (Functor m, Applicative m, MonadSupply m, MonadError MultipleErrors m) => Ident -> [([Binder], Either [(Guard, Expr)] Expr)] -> m Declaration
 makeCaseDeclaration ident alternatives = do
   let argPattern = length . fst . head $ alternatives
-  args <- map Ident <$> replicateM argPattern freshName
+  args <- mapM argName (foldl1 (zipWith resolveIdents) namedArgs)
```

Comment:
It might be worth extracting the expression on the right into a `let` for readability. I had to stare at it for a minute to see what it was doing.